HipercowApiUnitTests/Tools/ClusterHandleCacheTests.cs

```diff
@@ -0,0 +1,32 @@
+﻿// Copyright (c) Imperial College London. All rights reserved.
+
+namespace HipercowApiUnitTests.Tools
+{
+    using HipercowApi.Tools;
+    using Microsoft.Hpc.Scheduler;
+    using Moq;
+
+    /// <summary>
+    /// Tests on the unit class.
+    /// </summary>
+    public class ClusterHandleCacheTests
+    {
+        /// <summary>
+        /// Cached scheduler object is reused.
+        /// </summary>
+        [Fact]
+        public void CachedScheduler_Works()
+        {
+            var mockScheduler = new Mock<IScheduler>();
+            mockScheduler.Setup(x => x.Connect("potato")).Verifiable();
+            var mockSchedulerFactory = new Mock<ISchedulerFactory>();
+            mockSchedulerFactory.Setup(x => x.NewScheduler()).Returns(mockScheduler.Object);
+
+            ClusterHandleCache chc = new ClusterHandleCache(mockSchedulerFactory.Object);
+            IScheduler? sch1 = chc.GetClusterHandle("wpia-hn");
+            IScheduler? sch2 = chc.GetClusterHandle("wpia-hn");
+            Assert.NotNull(sch1);
+            Assert.Equal(sch1, sch2);
```

Comment:
Could you do this, to test that the scheduler returned actually is the mock scheduler?
```suggestion
            Assert.Equal(sch1, mockScheduler.Object);
            Assert.Equal(sch1, sch2);
```

It would also be good to test that NewScheduler has only been called once - not sure what the behaviour of `Returns` is, would it always return the given object or only on the first call? If the former, then if the cache had wrongly created a new scheduler on the second call, rather than returning the cached one, this test would still pass. I think you can do:
```
mockSchedulerFactory.Verify(x => x.NewScheduler(), Times.Once());
```

Reply:
Yep, that's better - I had a moment of doubt that it might have already been initialised, but that's in Program.cs which the unit tests aren't calling, so the above also still passes the tests!